libra_toolbox/neutron_detection/activation_foils/naa.py

```diff
@@ -0,0 +1,733 @@
+
+    if plot:
+        colors = ['red', 'green', 'magenta', 'brown', 'cyan']
+        fig, ax = plt.subplots(figsize=[5,4])
+        ax.stairs(hist, bins)
+        ax.plot(xvals, gauss(xvals, *parameters))
+
+        erg_str = ''
+
+        for i in range(len(peak_ergs)):
+            # plot dashed vertical lines at edges of peak
+            # ax.plot([xvals[peak_starts[i]]]*2, [0, np.max(hist)], '--', color=colors[i])
+            # ax.plot([xvals[peak_ends[i]]]*2, [0, np.max(hist)], '--', color=colors[i])
+
+            trap_cutoff_ys = parameters[0] + parameters[1] * xvals[peak_starts[i]:peak_ends[i]]
+            # print('Plot gauss params: {}'.format(all_peak_params[i]))
+            ax.fill_between(xvals[peak_starts[i]:peak_ends[i]], gauss(xvals[peak_starts[i]:peak_ends[i]], *all_peak_params[i]), 
+                        trap_cutoff_ys, color=colors[i], alpha=0.5)
+            erg_str += ' {:.0f},'.format(peak_ergs[i])
+
+        ax.set_ylabel('Counts', fontsize=14)
+        ax.set_ylim(-np.max(peak_amplitudes)*0.2, np.max(peak_amplitudes)*1.5)
+        ax.set_xlabel('Energy [keV]', fontsize=14)
+        plt.xticks(fontsize=12)
+        plt.yticks(fontsize=12)
+        ax.set_title('Peak:{} keV'.format(erg_str), fontsize=14)
+    
+    return areas
+
+
+
+def group_close_values(data, threshold=200):
+    # Sort the data to group values sequentially
+    data.sort()
+    
+    # Initialize groups and a temporary group
+    groups = []
+    temp_group = [data[0]]
+    
+    for i in range(1, len(data)):
+        # Check if the current value is within the threshold of the last value in the temp group
+        if abs(data[i] - temp_group[-1]) < threshold:
+            temp_group.append(data[i])
+        else:
+            # Commit the temp group to groups and start a new group
+            groups.append(tuple(temp_group))
+            temp_group = [data[i]]
+    
+    # Add the last group
+    groups.append(tuple(temp_group))
+    
+    return groups
+
+
+def get_peak_areas(hist, bins, peak_ergs, 
+                   overlap_width=200, search_width=400, plot=False):
+
+    areas = []
+
+    # organize peak energies into tuples, in which peak energies close enough
+    # to have overlapping peaks will be paired together
+    erg_groups = group_close_values(peak_ergs, threshold=overlap_width)
+    # print(erg_groups)
+
+    for erg_group in erg_groups:
+            areas += get_multipeak_area(hist, bins, erg_group, 
+                                        search_width=len(erg_group)*search_width,
+                                        plot=plot)
+            # print(areas)
+    return areas
+
+
+def energy_efficiency(counts, decay_lines, nuclides=None,
+                      plot_eff=True, plot_areas=False,
+                      overlap_width=200, search_width=400,
+                      degree=2, count_sum_peak=False,
+                      ax_eff=None):
+    if nuclides is None:
+        nuclides = decay_lines.keys()
```

Comment:
Just FYI you can write

```suggestion
    nuclides = nuclides or decay_lines.keys()
```